src/metabase/sync/analyze/fingerprint/fingerprinters.clj

```diff
@@ -157,12 +158,14 @@
               {:type {~(first field-type) fingerprint#}})))
          (trs "Error generating fingerprint for {0}" (sync-util/name-for-logging field#))))))
 
+(declare ->temporal)
+
 (defn- earliest
   ([] nil)
   ([acc]
    (some-> acc u.date/format))
   ([acc t]
-   (if (and t acc (t/before? t acc))
+   (if (and t acc (t/before? t (->temporal acc)))
```

Comment:
I don't think this is a good solution -- if I change the order from string, number to number, string it breaks. I think it would be better to change implementations of `ITemporalCoerceable ` so that they always return the same thing. Something like: `Long     (->temporal [this] (->temporal (t/instant this)))`